### PR TITLE
test: update tests for envOverrides

### PR DESCRIPTION
**Pull request by razvan**
# Description

Part of : https://github.com/stackabletech/issues/issues/548

:green_circle: Test:

```
--- PASS: kuttl (147.39s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/smoke_hadoop-3.4.0_zookeeper-3.9.2_zookeeper-latest-3.9.2_number-of-datanodes-2_datanode-pvcs-2hdd-1ssd_listener-class-external-unstable_openshift-false (147.37s)
PASS
```

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] CRD documentation for all fields, following the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs/style-guide).
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
- [ ] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] Code contains useful logging statements
- [x] (Integration-)Test cases added
- [ ] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs/style-guide).
- [ ] Changelog updated
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
- [ ] [Roadmap](https://github.com/orgs/stackabletech/projects/25/views/1) has been updated
```

### CODE REVIEW

**Review by adwk67 — DISMISSED** [review]
LGTM

File: tests/templates/kuttl/smoke/30-assert.yaml.j2 [hunk 1]
```diff
@@ -2,6 +2,22 @@
 apiVersion: kuttl.dev/v1beta1
 kind: TestAssert
 timeout: 600
+commands:
```
Comment (adwk67) [inline tests/templates/kuttl/smoke/30-assert.yaml.j2]: just one nit: if this is moved to a following step (31-assert), then the log output is much cleaner.

Comment (razvan) [reply tests/templates/kuttl/smoke/30-assert.yaml.j2]: Indeed. Done in https://github.com/stackabletech/hdfs-operator/pull/576/commits/e6bb0d24d3aff7575c09b7180698fcaeebd2c54b

Comment (adwk67) [reply tests/templates/kuttl/smoke/30-assert.yaml.j2]: Thanks!

**Review by adwk67 — APPROVED** [review]

**Pull request merged**